python/tank_vendor/shotgun_base/utils.py

```diff
@@ -18,6 +18,34 @@
 from .log import get_shotgun_base_logger
 log = get_shotgun_base_logger()
 
+
+def with_cleared_umask(func):
```

Comment:
Could this be a wrapper around a more generic mask decorator that would accept a value to mask with? we could then use this with the auth framework?